### ISSUE TITLE
priority making loading local xxx.safetensor natively supported in pipelines.

### ISSUE THREAD

**Issue by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [huggingface/diffusers#3856]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when we need to wait for the conversion when using safetensors model.
The common api xpipe.from_pretrained("xx.safetensors",use_safetensors=True) dont work.
The parser assumes that its a repo from huggingfacehub or h5 model type and throws>
FileNotFoundError: [Errno 2] No such file or directory: 'cachedir\\models--xxx--xxx.safetensors\\refs\\main'
so we are force to use from_ckpt()

**Additional context**
The safetensor project is also from huggingface org.It be made as a substandard to loading models.
Currently we need to use conversion and its very slow it makes the usage of safetensor less meaningful.
eg
____________
xxpipeline.from_ckpt()

or using conversion as suggested in https://github.com/huggingface/diffusers/issues/3732

**Comment by patrickvonplaten** [comment]
Hey @cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw,

Can you please post a code snippet of the requested feature you would like to see supported?
Note that we are by default always downloading safetensors already when using `.safetensors`

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
it probably will not be as simple as a snippet.If my concept of this library is not wrong as a user.

but in a badly simplified implementation we probably have something like.
if (notexist(path+"/config.json")):
	nativelyloadsafetensor(path,kwags)#Broken/unimplemented now
	if(model==None):loadsafetensorconvertdiffuser(path)#has lots of bug,like not respect cache_dir i fixed it,but havent pr
else:
	if(onlylocal):
		nativelyloadsafetensor(path,kwags)#Broken/unimplemented now
		if(model==None):loadsafetensorconvertdiffuser(path)#has lots of bug,like not respect cache_dir i fixed it,but havent pr
	else:
		tryload_else_download(path)#this is the only one 99% tested,but has the derived format problem.

From my experience,its not just about downloading safetensors* but loading local models should also be perfectly supported.
Secondly there is also the problem of derived/modified* format.

for example.
this project claims that it is diffuser compatible,https://huggingface.co/stabilityai/sdxl-vae
But in my experience here is the result of using common api
__________
customvae = AutoencoderKL.from_pretrained(vaemodel,cache_dir="stbldiff",use_safetensors=True)
>Entry Not Found for url: https://huggingface.co/stabilityai/sdxl-vae/resolve/main/config.json.

It is expecting more than that safetensor,not even stabilityai developers expected this.hahaha,so funny.

Franky,It is a very horrible implementation.
In layman analogy:
you get a nice car but in your country you need an extra poem to config the car then unlock then driving it.
But for others,just directly unlock car and drive.

user also has from_ckpt() but it is like a inconsistent level 2*support fallback function for safetensor/pytorch.bin
__________
    customvae = AutoencoderKL.from_ckpt(vaemodel,cache_dir="stbldiff",use_safetensors=True)
AttributeError: type object 'AutoencoderKL' has no attribute 'from_ckpt'

Ideally,
In best case scenario,
xxx.from_pretrained(vaemodel,cache_dir="stbldiff",use_safetensors=True)
would support loading
_____
h5 online downloaded.
diffuser style safetensor with extra config
direct safetensor only with conversion to diffuser format(This is available for some pipeline only!)
direct safetensor only natively(aka.just like the original project promised specification,This is not available now!)

however I also know it would be quite hard to actually implement all given diffuser idea is to not use common DRY abstraction?
so in the very least.
xxx.from_ckpt(vaemodel,cache_dir="stbldiff",use_safetensors=True)
should be implemented in all pipeline and making it possible to load
_________
direct safetensor only with conversion to diffuser format(This is available for some pipeline only!)
direct safetensor only natively(aka.just like the original project promised specification,This is not available now!)


OR
diffusers support loading object/parameters loaded from safetensor library and just add some link in each documentation for supported stuff.

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
supported safetensor,ckpt,pt apis matrix>
txt2img
_______
pt:ok
ckpt:
safetensor:(only support converted)

instructpix2pix
_____
pt:?
ckpt:?
safetensor:x

AutoencoderKL
_______
pt:ok
ckpt:x
safetensor:x

LoRA
_____
pt:?
ckpt:?
safetensor:ok(native?)

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
re bump

**Comment by patrickvonplaten** [comment]
@cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw I still don't understand the issue here. Also it would be nice that if you have a feature request to mark it as a feature request
